kotlinx-coroutines-core/jvm/src/scheduling/CoroutineScheduler.kt

```diff
@@ -533,6 +531,11 @@ internal class CoroutineScheduler(
         return ADDED_REQUIRES_HELP
     }
 
+    private fun currentWorker(): Worker? {
+        val worker = Thread.currentThread() as? Worker ?: return null
+        return if (worker.scheduler == this) worker else null
+    }
+
```

Comment:
Consider using Kotlin idioms here and highlighting the fact it is a simple function (expression):
```
private fun currentWorker(): Worker? =
    (Thread.currentThread() as? Worker)?.takeIf { it.scheduler == this }
```